source/server/filter_chain_manager_impl.cc

```diff
@@ -527,14 +560,31 @@ const Network::FilterChain* FilterChainManagerImpl::findFilterChainForApplicatio
   for (const auto& application_protocol : socket.requestedApplicationProtocols()) {
     const auto application_protocol_match = application_protocols_map.find(application_protocol);
     if (application_protocol_match != application_protocols_map.end()) {
-      return findFilterChainForSourceTypes(application_protocol_match->second, socket);
+      return findFilterChainForDirectSourceIP(*application_protocol_match->second.second, socket);
     }
   }
 
   // Match on a filter chain without application protocol requirements.
   const auto any_protocol_match = application_protocols_map.find(EMPTY_STRING);
   if (any_protocol_match != application_protocols_map.end()) {
-    return findFilterChainForSourceTypes(any_protocol_match->second, socket);
+    return findFilterChainForDirectSourceIP(*any_protocol_match->second.second, socket);
+  }
+
+  return nullptr;
+}
+
+const Network::FilterChain* FilterChainManagerImpl::findFilterChainForDirectSourceIP(
+    const DirectSourceIPsTrie& direct_source_ips_trie,
+    const Network::ConnectionSocket& socket) const {
+  auto address = socket.addressProvider().directRemoteAddress();
+  if (address->type() != Network::Address::Type::Ip) {
+    address = fakeAddress();
```

Comment:
To test my understanding, this is just a placeholder address so the matching code won't crash and will always fail the match (if a match is requested) right?
I know it's not your code but if I understand it right can you update the comments at fakeAddress to expand UDS -> unix domain socket and note it's just creating an address which should never match?

Also do we have a unit test for this?

Reply:
Yes, your understanding is correct. Nearly all of this code is copy/pasted from the SourceIP matching. I'll add comments and make sure there's a test for this.